src/python/pants/backend/project_info/rules/fast_dependencies_test.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from typing import List, Optional
+
+from pants.backend.project_info.rules import fast_dependencies
+from pants.backend.python.targets.python_library import PythonLibrary
+from pants.build_graph.build_file_aliases import BuildFileAliases
+from pants.testutil.console_rule_test_base import ConsoleRuleTestBase
+
+
+class FastDependenciesTest(ConsoleRuleTestBase):
+  goal_cls = fast_dependencies.Dependencies
+
+  @classmethod
+  def alias_groups(cls) -> BuildFileAliases:
+    return BuildFileAliases(
+      targets={
+        "python_library": PythonLibrary,
+      },
+    )
+
+  @classmethod
+  def rules(cls):
+    return super().rules() + fast_dependencies.rules()
+
+  def create_python_library(
+    self, path: str,
+    *,
+    dependencies: Optional[List[str]] = None
+  ) -> None:
+    self.create_library(
+      path=path,
+      target_type="python_library",
+      name="target",
+      sources=[],
+      dependencies=dependencies or []
+    )
+
+  def assert_dependencies(self, *, target: str, expected: List[str], transitive: bool = True) -> None:
+    args = ["--no-transitive"] if not transitive else []
+    self.assert_console_output(*expected, args=args + [target])
+
+  def test_no_dependencies(self):
+    self.create_python_library(path="some/target")
+    self.assert_dependencies(
+      target="some/target",
+      expected=[],
+      transitive=False,
+    )
```

Comment:
For filedeps test, we tested what happens with no target. I think it's worth adding here to ensure we no-op and don't raise any exception, for example?

https://github.com/pantsbuild/pants/blob/5fcb49245701c85243e49e981d6478df3571e9a2/src/python/pants/rules/core/filedeps_test.py#L63-L64